Makefile

```diff
@@ -156,7 +156,14 @@ rpm: build ## build the rpm packages
 run: build ## run the docker daemon in a container
 	$(DOCKER_RUN_DOCKER) sh -c "KEEPBUNDLE=1 hack/make.sh install-binary run"
 
-shell: build ## start a shell inside the build env
+shell: ## start a shell inside the build env
+	docker build \
+		${BUILD_APT_MIRROR} \
+		${DOCKER_BUILD_ARGS} \
+		-t "$(DOCKER_IMAGE)" \
+		--target dev-interactive \
+		-f "$(DOCKERFILE)" .
+	# TODO: always set BINDDIR=.
```

Comment:
I'm not sure how to make this work. The current logic for BINDDIR is a bit messy. 

Do we really need separate bind mount for `.git` and `bundles` ?

Reply:
Since the Makefile is not used by CI, can we default to `-v $PWD:/go/src/github.com/docker/docker` ?

Reply:
to skip uploading the build context, we could pipe the Dockerfile;

```
cat Dockerfile | docker build -
```

Reply:
and yes, the bind_dir is messy, think I linked to an old PR of mine from the multi-stage build PR (typing from my phone right now :sweat_smile:)

Reply:
We could also just set it to some empty directory, but we still need the context for `hack/dockerfile` so I don't think we can do either.

Reply:
Ah, yes, I recall now, that was the thing I ran into at the time (and had a dirty workaround to dynamically change the `.dockerignore`)

Reply:
We need cookies to bribe Tonis in accepting https://github.com/moby/moby/issues/12886